AI/Nullkiller/Behaviors/BuyArmyBehavior.cpp

```diff
@@ -57,8 +57,7 @@ Goals::TGoalVec BuyArmyBehavior::decompose() const
 				continue;
 			}
 
-			if(ai->nullkiller->heroManager->getHeroRole(targetHero) == HeroRole::MAIN
-				&& targetHero->getArmyStrength() >= 300)
```

Comment:
Removed a hack. A way more complicated logic is actually needed here

Reply:
am I correctly understand that now all army will be consolidated under single hero?

Reply:
No, but this thing prevented AI from buying army from town. The idea was to prefer types of army suitable for main hero. This needs better logic.